src/main/java/com/marklogic/kafka/connect/source/RowBatcherSourceTask.java

```diff
@@ -90,10 +93,11 @@ protected void performPoll(QueryHandler queryHandler, List<SourceRecord> newSour
             rowBatcher.awaitCompletion();
             dataMovementManager.stopJob(rowBatcher);
 
-            if (constraintColumn != null && !newSourceRecords.isEmpty()) {
+            if (constraintValueStore != null && !newSourceRecords.isEmpty()) {
                 long queryStartTimeInMillis = rowBatcher.getServerTimestamp();
-                previousMaxConstraintColumnValue = queryHandler.updatePreviousMaxConstraintColumnValue(queryStartTimeInMillis);
-                logger.info("Storing new max constraint value: " + previousMaxConstraintColumnValue);
+                String newMaxConstraintColumnValue = queryHandler.updatePreviousMaxConstraintColumnValue(queryStartTimeInMillis);
+                logger.info("Storing new max constraint value: " + newMaxConstraintColumnValue);
+                constraintValueStore.storeConstraintState(newMaxConstraintColumnValue, newSourceRecords.size());
```

Comment:
I did a test here where I forced a failure by setting the uri property to "hi.xml" - ML complains because it assumes I want an XML document. But it could fail for other reasons too, like missing an update permissions.

The error then gets logged, stating "job failed to complete successfully" - but the source records that were retrieved are still returned. That would be surprising to a user - if the job failed (though it didn't, the job succeeded), I think a user would expect no data to be returned. 

I think that suggests the following changes:

1. The try/catch should only be around this block of code for getting and storing the max value
2. The error message should be specific to that goal and should be a warning instead of an error since we're still going to return data - i.e. "Unable to update max constraint value, cause: " + e.getMessage()